File: VB/LookUpEditWithHints/LookUpEditHints.vb
Imports System
Imports DevExpress.XtraEditors
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Popup

Namespace LookUpEditWithHints

    Friend Class LookUpEditHints
        Inherits LookUpEdit

        Shared Sub New()
            RepositoryItemLookUpEditHints.RegisterLookUpEditHints()
        End Sub

        Public Sub New()
            MyBase.New()
        End Sub

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return RepositoryItemLookUpEditHints.LookUpEditHintsName
            End Get
        End Property

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As RepositoryItemLookUpEditHints
            Get
                Return TryCast(MyBase.Properties, RepositoryItemLookUpEditHints)
            End Get
        End Property

        Protected Overrides Function CreatePopupForm() As PopupBaseForm
            Return New PopupLookUpEditHintsForm(Me)
        End Function

        Public Property DescriptionField As String
            Set(ByVal value As String)
                Properties.DescriptionField = value
                OnPropertiesChanged()
            End Set

            Get
                Return Properties.DescriptionField
            End Get
        End Property

        Public Custom Event BeforeShowingTooltip As EventHandler
            AddHandler(ByVal value As EventHandler)
                AddHandler Properties.BeforeShowingTooltip, value
            End AddHandler

            RemoveHandler(ByVal value As EventHandler)
                RemoveHandler Properties.BeforeShowingTooltip, value
            End RemoveHandler

            RaiseEvent(ByVal sender As Object, ByVal e As EventArgs)
            End RaiseEvent
        End Event
    End Class
End Namespace

File: CS/LookUpEditWithHints/LookUpEditHints.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors;
using System.ComponentModel;
using DevExpress.XtraEditors.Popup;

namespace LookUpEditWithHints
{
    class LookUpEditHints : LookUpEdit
    {
        static LookUpEditHints() { RepositoryItemLookUpEditHints.RegisterLookUpEditHints(); }
        public LookUpEditHints() : base() { }

        public override string EditorTypeName
        { get { return RepositoryItemLookUpEditHints.LookUpEditHintsName; } }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemLookUpEditHints Properties
        { get { return base.Properties as RepositoryItemLookUpEditHints; } }

        protected override DevExpress.XtraEditors.Popup.PopupBaseForm CreatePopupForm()
        {
            return new PopupLookUpEditHintsForm(this);
        }

        public string DescriptionField {
            set { Properties.DescriptionField = value; OnPropertiesChanged(); }
            get { return Properties.DescriptionField; }
        }

        public event EventHandler BeforeShowingTooltip
        {
            add { Properties.BeforeShowingTooltip += value; }
            remove { Properties.BeforeShowingTooltip -= value; }
        }
    }
}
